code/game/gamemodes/cult/cult_structures.dm

```diff
@@ -59,6 +79,12 @@
 			to_chat(user, "You hit the pylon!")
 		playsound(get_turf(src), 'sound/effects/Glasshit.ogg', 75, 1)
 
+/obj/structure/cult/pylon/attack_animal(mob/living/simple_animal/M)
+	if(istype(M, /mob/living/simple_animal/construct/builder))
+		if(isbroken && prob(20))
+			repair(M)
```

Comment:
I disagree, artificers already have a spell for that, I purposely let them able to attack pylons so they wouldn't accidentally trap themselves. (plus why a 20% chance anyway?)